sdk/tables/aztables/table_client.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package aztables
+
+import (
+	"context"
+	"errors"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/runtime"
+)
+
+// A TableClient represents a URL to an Azure Storage blob; the blob may be a block blob, append blob, or page blob.
+type TableClient struct {
+	client  *tableClient
+	service *TableServiceClient
+	cred    SharedKeyCredential
+	name    string
+}
+
+type TableUpdateMode string
+
+const (
+	Replace TableUpdateMode = "replace"
+	Merge   TableUpdateMode = "merge"
+)
+
+// NewTableClient creates a TableClient object using the specified URL and request policy pipeline.
+func NewTableClient(tableName string, serviceURL string, cred azcore.Credential, options *TableClientOptions) (*TableClient, error) {
+	s, err := NewTableServiceClient(serviceURL, cred, options)
+	return s.GetTableClient(tableName), err
+}
+
+func (t *TableClient) Name() string {
```

Comment:
If this is only needed for testing, I think this should be unexported. If customer's are going to need it, then you should probably just export the field on the struct, that is more Go-ish.

Reply:
It's a feature we got feedback on for the .NET client. I'll move it to a field.